app/validators/immigration_status_validator.rb

```diff
@@ -0,0 +1,25 @@
+class ImmigrationStatusValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, application_choice)
+    return if did_not_add_nationality_yet?(application_choice) || immigration_right_to_work?(application_choice) || course_can_sponsor_visa?(application_choice)
```

Comment:
We don't have a limit on line length, but could you split these conditions into multiple lines? You are doing it below.

Reply:
Done!